src/Ng/Jobs/Catalog2IconJob.cs

```diff
@@ -0,0 +1,90 @@
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NuGet.Protocol.Catalog;
+using NuGet.Services.Configuration;
+using NuGet.Services.Metadata.Catalog;
+using NuGet.Services.Metadata.Catalog.Icons;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace Ng.Jobs
+{
+    public class Catalog2IconJob : LoopingNgJob
+    {
+        private const int DegreeOfParallelism = 100;
+        private IconsCollector _collector;
+        private DurableCursor _front;
+
+        public Catalog2IconJob(ITelemetryService telemetryService, ILoggerFactory loggerFactory)
+            : base(telemetryService, loggerFactory)
+        {
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            ServicePointManager.DefaultConnectionLimit = DegreeOfParallelism;
+
+            var verbose = arguments.GetOrDefault(Arguments.Verbose, false);
+            var packageStorageBase = arguments.GetOrThrow<string>(Arguments.ContentBaseAddress);
+            var auxStorageFactory = CreateAuxStorageFactory(arguments, verbose);
+            var targetStorageFactory = CreateTargetStorageFactory(arguments, verbose);
+            var packageStorage = new AzureStorage(new Uri(packageStorageBase), TimeSpan.FromMinutes(15), TimeSpan.FromMinutes(10), false, false, true, null);
+            var source = arguments.GetOrThrow<string>(Arguments.Source);
+            var iconProcessor = new IconProcessor(TelemetryService, LoggerFactory.CreateLogger<IconProcessor>());
+            var httpHandlerFactory = CommandHelpers.GetHttpMessageHandlerFactory(TelemetryService, verbose);
+            var httpMessageHandler = httpHandlerFactory();
+            var httpClient = new HttpClient(httpMessageHandler);
+            var simpleHttpClient = new SimpleHttpClient(httpClient, LoggerFactory.CreateLogger<SimpleHttpClient>());
+            var catalogClient = new CatalogClient(simpleHttpClient, LoggerFactory.CreateLogger<CatalogClient>());
+            var httpResponseProvider = new HttpResponseMessageProvider(httpClient);
+            var externalIconProvider = new ExternalIconContentProvider(httpResponseProvider, LoggerFactory.CreateLogger<ExternalIconContentProvider>());
+            var iconCopyResultCache = new IconCopyResultCache(auxStorageFactory.Create());
+
+            var leafProcessor = new CatalogLeafDataProcessor(
+                packageStorage,
+                iconProcessor,
+                externalIconProvider,
+                iconCopyResultCache,
+                TelemetryService,
+                LoggerFactory.CreateLogger<CatalogLeafDataProcessor>());
+
+            _collector = new IconsCollector(
+                new Uri(source),
+                TelemetryService,
+                targetStorageFactory,
+                catalogClient,
+                leafProcessor,
+                iconCopyResultCache,
+                () => httpMessageHandler,
```

Comment:
This causes crashes when job reaches the end of the catalog, because base collector disposes of HttpClient at that point, which in turn disposes of MessageHandler.